lib/mongoid/association/referenced/has_many/proxy.rb

```diff
@@ -217,9 +217,23 @@ def each(&block)
           # @example Are there persisted documents?
           #   person.posts.exists?
           #
+          # @param [ :none | nil | false | Hash | Object ] id_or_conditions
+          #   When :none (the default), returns true if any persisted
+          #   documents exist in the association. When nil or false, this
+          #   will always return false. When a Hash is given, this queries
+          #   the documents in the association for those that match the given
+          #   conditions, and returns true if any match. Any other argument is
+          #   interpreted as an id, and queries for the existence of documents
+          #   in the association with a matching _id.
+          #
           # @return [ true | false ] True is persisted documents exist, false if not.
-          def exists?
-            criteria.exists?
+          def exists?(id_or_conditions = :none)
+            case id_or_conditions
+            when :none then criteria.exists?
+            when nil, false then false
+            when Hash then criteria.where(id_or_conditions).exists?
+            else criteria.where(_id: id_or_conditions).exists?
+            end
```

Comment:
why can't this delegate to criteria#exists?

Reply:
Good catch! It certainly could. I'll fix.